sdgym/benchmark.py

```diff
@@ -253,157 +267,139 @@ def _run_on_dask(jobs, verbose):
-            from the specified bucket.
-        aws_secret (str):
-            If an ``aws_secret`` is provided, the given secret access key will be used to read
-            from the specified bucket.
-        jobs (list[tuple]):
-            List of jobs to execute, as a sequence of tuple-like objects containing synthesizer,
-            dataset and iteration-id specifications. If not passed, the jobs list is build by
-            combining the synthesizers and datasets given instead.
-        max_rows (int):
-            Cap the number of rows to model from each dataset. Rows will be selected in order.
-        max_columns (int):
-            Cap the number of columns to model from each dataset.
-            Columns will be selected in order.
+        multi_processing_config (dict or ``None``):
+            The config to use if multi-processing is desired. For example,
+            {
+             'package_name': 'dask' or 'multiprocessing',
+             'num_workers': 4
+            }
 
     Returns:
         pandas.DataFrame:
-            A table containing one row per synthesizer + dataset + metric + iteration.
+            A table containing one row per synthesizer + dataset + metric.
     """
-    if cache_dir and not is_s3_path(cache_dir):
-        cache_dir = Path(cache_dir)
-        os.makedirs(cache_dir, exist_ok=True)
+    if detailed_results_folder and not is_s3_path(detailed_results_folder):
+        detailed_results_folder = Path(detailed_results_folder)
+        os.makedirs(detailed_results_folder, exist_ok=True)
 
-    run_id = os.getenv('RUN_ID') or str(uuid.uuid4())[:10]
+    max_rows = None
+    max_columns = None
+    if limit_dataset_size:
+        max_rows = 1000
+        max_columns = 10
 
-    if workers == -1:
-        num_gpus = get_num_gpus()
-        if num_gpus > 0:
-            workers = num_gpus
-        else:
-            workers = multiprocessing.cpu_count()
-
-    if jobs is None:
-        synthesizers = get_synthesizers(synthesizers)
-        datasets = get_dataset_paths(datasets, datasets_path, bucket, aws_key, aws_secret)
+    run_id = os.getenv('RUN_ID') or str(uuid.uuid4())[:10]
 
-        job_tuples = list()
-        for dataset in datasets:
-            for synthesizer in synthesizers:
-                for iteration in range(iterations):
-                    job_tuples.append((synthesizer, dataset, iteration))
+    synthesizers = get_synthesizers(synthesizers)
+    datasets = get_dataset_paths(sdv_datasets, None, None, None, None)
 
-    else:
-        job_tuples = list()
-        for synthesizer, dataset, iteration in jobs:
-            job_tuples.append((
-                get_synthesizers([synthesizer])[0],
-                get_dataset_paths([dataset], datasets_path, bucket, aws_key, aws_secret)[0],
-                iteration
-            ))
+    job_tuples = list()
+    for dataset in datasets:
+        for synthesizer in synthesizers:
+            job_tuples.append((synthesizer, dataset, 1))
```

Comment:
any reason not to filter for modalities here? Then you wouldn't need an if statement on line 356

Reply:
One reason is so that we won't need to call `load_dataset` in two places.